### ISSUE TITLE
Improve popup content scrollview interaction

### ISSUE THREAD

**Issue by willsmillie** [LeoNatan/LNPopupController#572]
I'm trying to create a video player queue using LNPopupUI. I have it working almost perfectly, except for the popup interaction gesture recognizer's drag-to-dismiss functionality, which only works when grabbing the handle (as opposed to anywhere on the popup's content view). If I disable the scroll view interactions, I can swipe down again, but this obviously breaks the scroll view paging.

A number of years ago, with the Objective-C version, I was able to use a combination of `UIGestureRecognizerDelegate` along with iterating through views and setting some `gesture.delaysTouchesBegan` / `scrollView.delaysContentTouches` values, but after a few days of trying, I have been unable to achieve similar results.

My setup is as follows:

I have a root SwiftUI view with a `LazyVGrid` containing buttons for each video. Tapping a video adds it to a `videoPlayerManager` and presents the popup bar.

In the popup bar, I have a `UIHostingController` which wraps a horizontally scrolling `UIPageViewController`, into which the `videoPlayerManager` is injected, and coordinates the children view/controllers.

Here is one of the simpler approaches I attempted, which oddly enough, the gesture delegate doesn't seem to be assigned:

```
class VideoPageViewController: UIPageViewController, UIPageViewControllerDataSource, UIPageViewControllerDelegate, UIGestureRecognizerDelegate {

  override func viewDidAppear(_ animated: Bool) {
    super.viewDidAppear(animated)

    self.popupContentView.popupInteractionGestureRecognizer.delegate = self
    self.popupContentView.popupInteractionGestureRecognizer.delaysTouchesBegan = false
    self.popupContentView.popupInteractionGestureRecognizer.cancelsTouchesInView = false
    self.popupContentView.popupInteractionGestureRecognizer.requiresExclusiveTouchType = false

    self.scrollView!.delaysContentTouches = true
    self.scrollView!.panGestureRecognizer.delaysTouchesBegan = true
    self.scrollView!.panGestureRecognizer.require(toFail: self.popupContentView.popupInteractionGestureRecognizer)
    self.scrollView!.panGestureRecognizer.shouldRequireFailure(of: self.popupContentView.popupInteractionGestureRecognizer)
  }

  func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWith otherGestureRecognizer: UIGestureRecognizer) -> Bool {
      return true
  }
}
```
Any insights or pointers would be greatly appreciated! Let me know if you'd like me to provide a sample project. Thanks for the stellar library!

**Comment by LeoNatan** [comment]
Please try with LNPopupController 2.19.3 (if using LNPopupUI, update your packages to the latest versions).

**Comment by willsmillie** [comment]
Seems to be working great! Looks like there is some tweaking to be had on my end when adding an AVPlayer, but seems like #251 has some answers! Thanks so much Leo!

**Comment by LeoNatan** [comment]
Great find!

**Comment by LeoNatan** [comment]
Hello,
I’ve released 2.19.10 with further improvements to scrolling content inside the popup. I would appreciate if you can check with your project that nothing broke and that everything is behaving as it should after the update. It should work, but `UIPageViewController` and its internal queueing scroll view are their own kind of hell, so who knows. 🤣

I think the system is in a good state right now. The only scenario that is not supported is multidirectional scroll without isDirectionalLockEnabled = true, in which case the system gives up and will not attempt to close the popup. I think this is a good compromise, as it is not really clear what is the most desired behavior anyway. And the user can implement the delegate and force one way or another. I also updated the documentation to explicitly state the behavior I want to achieve in all cases.
